docs/porting-example/steps/step_01_hpy_legacy.c

```diff
@@ -143,36 +143,47 @@ static HPyType_Spec Point_Type_spec = {
     .defines = point_defines,
 };
 
+// HPy supports only multiphase module initialization, so we must migrate the
+// single phase initialization by extracting the code that populates the module
+// object with attributes into a separate 'exec' slot. The module is not
+// created manually by calling API like PyModule_Create, but the runtime creates
+// the module for us from the specification in HPyModuleDef, and we can provide
+// additional slots to populate the module before its initialization is finalized
+HPyDef_SLOT(module_exec, HPy_mod_exec)
+static int module_exec_impl(HPyContext *ctx, HPy mod)
+{
+    HPy point_type = HPyType_FromSpec(ctx, &Point_Type_spec, NULL);
+    if (HPy_IsNull(point_type))
+        return -1;
+    HPy_SetAttr_s(ctx, mod, "Point", point_type);
+    return 0;
+}
+
 // Legacy module methods (the "dot" method is still a PyCFunction)
 static PyMethodDef PointModuleLegacyMethods[] = {
     {"dot", (PyCFunction)dot, METH_VARARGS, "Dot product."},
     {NULL, NULL, 0, NULL}
 };
 
-// HPy module methods (no methods have been ported yet)
+// HPy module methods: no regular methods have been ported yet,
+// but we add the module execute slot
 static HPyDef *module_defines[] = {
+    &module_exec,
     NULL
 };
 
 static HPyModuleDef moduledef = {
-    .name = "step_01_hpy_legacy",
+    // .name = "step_01_hpy_legacy",
+    // ^-- .name is not needed for multiphase module initialization,
+    // it is always taken from the ModuleSpec
```

Comment:
What happens if `.name` is provided anyway? Is this suggestion true?
```suggestion
    // it is always taken from the ModuleSpec. It can optionally be provided
    // here but must match
```

Reply:
There's no `name` field in the struct anymore in HPy. CPython always ignores `name` in `PyModuleDef` in multi-phase init. I assume they keep it there only for backwards binary compatibility (and single phase init).

Reply:
In that case maybe this should state "it can be provided here but will be ignored"?